### PR TITLE
Check if hosts are noticeable in slave scans

### PR DESCRIPTION
**Pull request by timopollmeier**
When hosts are added to the assets from a slave scan, perform the usual
checks of whether each host is alive and has results.

**Checklist**:

- Tests N/A
- [x] [CHANGELOG](https://github.com/greenbone/gvmd/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: src/manage_sql.c [hunk 1]
```diff
@@ -56371,6 +56380,16 @@ hosts_set_details (report_t report)
...
+       /* Ensure that every report host detail has a corresponding host
+        *  in the assets. */
+       " AND EXISTS (SELECT host"
+       "               FROM host_identifiers"
+       "              WHERE source_id = (SELECT uuid FROM reports"
+       "                            WHERE id = %llu)"
+       "                AND (SELECT name FROM hosts WHERE id = host)"
+       "                      = (SELECT host FROM report_hosts"
+       "                         WHERE id = report_host_details.report_host)"
+       "             LIMIT 1)"
```
Comment (mattmundell) [inline src/manage_sql.c]: I don't think LIMIT is ever considered inside EXISTS, so this line could be left out for neatness.

File: src/manage_sql.c [hunk 2]
```diff
@@ -56371,6 +56380,16 @@ hosts_set_details (report_t report)
        " AND (SELECT value = 'yes' FROM task_preferences"
        "      WHERE task = (SELECT task FROM reports WHERE id = %llu)"
        "      AND name = 'in_assets')"
+       /* Ensure that every report host detail has a corresponding host
+        *  in the assets. */
+       " AND EXISTS (SELECT host"
+       "               FROM host_identifiers"
+       "              WHERE source_id = (SELECT uuid FROM reports"
+       "                            WHERE id = %llu)"
```
Comment (mattmundell) [inline src/manage_sql.c]:
```suggestion
       "                                 WHERE id = %llu)"
```

File: src/manage_sql.c [hunk 3]
```diff
@@ -56371,6 +56380,16 @@ hosts_set_details (report_t report)
        " AND (SELECT value = 'yes' FROM task_preferences"
        "      WHERE task = (SELECT task FROM reports WHERE id = %llu)"
        "      AND name = 'in_assets')"
+       /* Ensure that every report host detail has a corresponding host
+        *  in the assets. */
+       " AND EXISTS (SELECT host"
```
Comment (mattmundell) [inline src/manage_sql.c]: No need to specify the column with EXISTS.

File: src/manage_sql.c (hunk 1 above)
Comment (timopollmeier) [reply src/manage_sql.c]: I've addressed this and the following points in 0435471b341bf9b81a156b6835928c51ea5dfdb8.

**Pull request merged**